frontend/src/app/shared/models/error.interceptor.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpHandler,
+  HttpInterceptor,
+  HttpErrorResponse,
+  HttpRequest,
+  HttpEvent,
+} from '@angular/common/http';
+import { throwError, Observable } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ErrorInterceptor implements HttpInterceptor {
```

Comment:
Please move the file under core/intercepters folder. Create core module and make it part of it. Then import this module to AppModule. Try to keep AppModule as clear as possible.